frontend/online-shop/src/App.tsx

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import "./App.css";
+import ProductList from "./components/ProductList";
+import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
+import Header from "./components/Header";
+import ProductDetails from "./components/ProductDetails";
+
+function App() {
+  return (
+    <Router>
+      <Header />
+      <Switch>
```

Comment:
What if the user is directly accessing a path that you don't have in your switch? For example http://localhost:4200/asdfg. For such cases you need to have a wildcard route that catches all the paths that do not fit with any other declared route and return a 404 page.